### ISSUE TITLE
Cancelling a navigation instruction with a redirect from the authorize pipeline step raises inconsistent navigation events

### ISSUE THREAD

**Issue by bfil** [aurelia/router#177]
I've tested the new navigation events (`processing` -> `success`|`error`|`canceled` -> `compete`) using the routing pipeline steps as described [in the documentation](http://aurelia.io/docs.html#customizing-the-navigation-pipeline), but the new router does not seem to raise consistent events when redirecting from a pipeline step.

When redirecting from `/welcome` to `/login` in the `authorize` pipeline step the `router:navigation:complete` event is fired with the `/welcome` instruction, no other events are raised in regards to the redirection/navigation to `/login`, but the redirection does happen (original example in PR https://github.com/aurelia/router/pull/97).

This causes issues when components subscribe to these events to change their appearance/behaviour, because the application ends up onto the `/login` page correctly, but the components receive the event telling them they are on the `/welcome` page.

**Comment by bfil** [comment]
I've found a workaround for the moment, which is to check `router.currentInstruction` from within the `router:navigation:complete` event handler, instead of using the event data.

**Comment by bryanrsmith** [comment]
It happens because events are raised using the original instruction, but I agree it's not right. We should raise events using the instruction that actually created that result.

**Comment by bryanrsmith** [comment]
Thanks for the reminder about 'cancelled'. I'll make sure we're using the American English inflection consistently.

**Comment by EisenbergEffect** [comment]
@bryanrsmith Can we close this now?

**Comment by bryanrsmith** [comment]
No, I need to make an adjustment to the instruction attached to the event.

**Comment by aozisik** [comment]
Any progress on this issue? I have also hit the same wall

**Comment by powerbuoy** [comment]
Had the same issue but using `event.instruction.router.currentInstruction.config.name` instead of `event.instruction.config.name`.

**Comment by danhowitt** [comment]
@powerbuoy +1 very useful.

**Comment by nexbit** [comment]
Any news on this issue? It is a major issue with the router, and renders router events pretty useless now..

**Comment by Alexander-Taran** [comment]
I wonder.. If it is still the same?

**Comment by RomkeVdMeulen** [comment]
@Alexander-Taran It is: just ran into this specific problem.
